users/src/main/java/io/crate/operation/auth/TrustAuthentication.java

```diff
@@ -29,12 +30,16 @@
 public class TrustAuthentication implements AuthenticationMethod {
 
     static final String NAME = "trust";
-    private static final String SUPERUSER = "crate";
+    private final UserManager userManager;
+
+    public TrustAuthentication(UserManager userManager) {
```

Comment:
Could consider to reducing the dependency to a BooleanSupplier for the exists check. The AuthenticationMethod doesn't need to be able to access create/drop user functionality